File: 238. Product of Array Except Self/src/main/Main.java
package main;

class Solution {
    public int[] productExceptSelf(int[] nums) {
    	double EPS = 1e-9;
    	double sum = 0;
    	for(int i = 0;i < nums.length;i++)
    		sum += Math.log10(nums[i]);
    	
    	for(int i = 0;i < nums.length;i++) {
    		nums[i] = (int)(EPS + Math.pow(10.00, sum - Math.log10(nums[i])));
    	}
    	return nums;
    }
}

public class Main {

	public static void main(String[] args) {
		//int []nums = {1,0};
		int []nums = {10, 3, 5, 6, 2};
		nums = new Solution().productExceptSelf(nums);
		for(int i : nums)
			System.out.print(i + " ");
		System.out.println();
	}
}

File: 238. Product of Array Except Self/main.cpp
#include <iostream>
#include <cstdlib>
#include <vector>
#include <cstring>
#include <algorithm>

using namespace std;

class Solution {
    public:
        vector<int> productExceptSelf(vector<int>& nums) {
            int temp = 1;
            vector<int> res(nums.size());
            
            fill(res.begin(), res.end(), 1);           
            for(int i = 0;i < res.size();i++) {
                res[i] = temp;
                temp *= nums[i];
            }
            
            temp = 1;
            
            for(int i = res.size()-1;i >= 0;i--) {
                res[i] *= temp;
                temp *= nums[i];
            }
            return res;
        }
};

int main(int argc, char** argv) {
    vector<int> vec = {1,2,3,4};
    vec = Solution().productExceptSelf(vec);
    for(auto& i : vec)
        cout << i << " ";
    cout << endl;
    return 0;
}